db/builder.cc

```diff
@@ -131,9 +131,17 @@ Status BuildTable(
       file->SetIOPriority(io_priority);
       file->SetWriteLifeTimeHint(write_hint);
 
-      file_writer.reset(
-          new WritableFileWriter(std::move(file), fname, file_options, env,
-                                 ioptions.statistics, ioptions.listeners));
+      if (ioptions.enable_sst_file_checksum &&
```

Comment:
What if `enable_sst_file_checksum && ioptions.sst_file_checksum == nullptr`? Looks like the default crc32 impl should be used.